### DISCUSSION TITLE
Option to remove guests from 'heading to ride' for specified rides/stalls

### DISCUSSION THREAD

**Discussion by astraylife1** [OpenRCT2/OpenRCT2#20752] [Feature requests]
For example, if you create a tower with various ride and stalls at the top but guests can only access it via an elevator, guests will try to "head for X" from the ground but are unable to find it. This create path crowding and generally poor park ratings.

It would be great if there was an option to check for specific rides/stalls to disable guests from seeking them out. Not sure if this would be better as a plugin but it's frustrating when you want to create areas that are only accessible via transport rides.

**Comment by ZehMatt** [comment]
We would need a more sophisticated way to easily query path connections which would also be helpful to improve the path finding for guests, @IntelOrca started working on a graph for paths that but no idea where its at. A work-around for this for the time being could be for guests to check the height difference, not optimal but might improve this situation a bit.

**Reply by astraylife1** [reply]
Yeah, that would be a helpful solution. The guests at the base are trying to get to the restroom and rides on the tower and the guests in the tower keep trying to get to the shops and restrooms at the base. A small change like what you suggested would make it more worthwhile to build observation towers without ruining park ratings.

**Comment by ocalhoun6** [comment]
Another possible -- though by no means *easy* -- solution is something that's been floated as a new feature here for a while: making guests (and maybe even staff) capable of pathfinding through transport rides (or possibly even non-transport rides with multiple stations).

Then, there would no longer be a problem because the guests *would* be able to find the stall/ride they were heading for.

It would definitely be quite difficult to implement, but if it could be done, it would be a *huge* improvement to the game, as well as completely solving this problem.

**Reply by ZehMatt** [reply]
It is actually not too difficult, we want to use a graph to build all connections so its just traversing path -> station -> tracks -> station -> path etc. The difficulty lies in rewriting the code actually, IntelOrca did start something like this as I mentioned before.
